pkg/server/handler/v1alpha1/workflowrun.go

```diff
@@ -375,13 +376,50 @@ func getContainerLogStream(tenant, project, workflow, workflowrun, stage string,
 	folderReader := stream.NewFolderReader(logFolder, prefix, exclusions, time.Second*10)
 	defer folderReader.Close()
 
-	err = websocketutil.Write(ws, folderReader)
+	ctx, cancel := context.WithCancel(context.Background())
+	go watchStageStatus(common.TenantNamespace(tenant), workflowrun, stage, cancel)
+	err = websocketutil.Write(ws, folderReader, ctx.Done())
 	if err != nil {
 		log.Error("websocket writer error:", err)
 	}
 	return err
 }
 
+// watchStageStatus watches status of a specific stage, when it is terminated, call the cancel func.
+func watchStageStatus(namespace, wfrName, stgName string, cancel context.CancelFunc) {
```

Comment:
It would be more robust to get and check wfr regularly, such as per 5min.

Reply:
ok

Reply:
`watchStageStatus `  --> `watchStageTermination`

Reply:
`cancel` --> `onTerminated`